### DISCUSSION TITLE
How to pass styled component as props to another element

### DISCUSSION THREAD

**Discussion by salimabsi** [styled-components/styled-components#3811] [Q&A] [answered]
I have component that copies the text passed to it - `<CopyText />`
![image](https://user-images.githubusercontent.com/75076979/190145828-d77a38fe-2a3d-42c3-9776-3a6df85b8c56.png)

When I invoke it, how to pass a styled component to it, or is there another way of doing it?
![image](https://user-images.githubusercontent.com/75076979/190146283-77c32b6d-0d6c-431f-a1db-91ef4ae5cc6c.png)

**Comment by auchynnikau** [comment]
@salimabsi  You can try some of this ways:

```js
const Parent = () => {
  return (
    <Child childComp={<YourCompToPass />} />
  )
};

const Child = (props) => { 
  return (
    <>
     {props.childComp}  
    </>
  )
}
```

Or using `props.children` property:
```js
const Parent = () => {
  return (
    <Child>
      <YourCompToPass />
    </Child
  )
};

const Child = (props) => { 
  return (
    <>
     {props.children}  
    </>
  )
}
```

**Comment by quantizor** [comment, marked as answer]
Styled components are regular React components, so you can pass them as props like any other component:

```jsx
<CopyText TextComponent={MyStyledText} />
```

Then render it inside `CopyText`:
```jsx
const CopyText = ({ TextComponent, children }) => (
  <TextComponent>{children}</TextComponent>
);
```

This is standard React composition and is not specific to styled-components.
